rero_mef/agents/viaf/api.py

```diff
@@ -255,49 +310,76 @@ def get_missing_agent_pids(cls, agent, verbose=False):
+            if verbose:
+                click.echo(f'Get pids from VIAF with {agent_pid_name} ...')
+            query = AgentViafSearch() \
+                .filter('bool', should=[Q('exists', field=agent_pid_name)]) \
+                .source(['pid', agent_pid_name])
+            progress = progressbar(
+                items=query.scan(),
+                length=query.count(),
+                verbose=verbose
+            )
+            pids_viaf = []
+            for hit in progress:
+                viaf_pid = hit.pid
+                agent_pid = hit.to_dict().get(agent_pid_name)
+                if pids_db.get(agent_pid):
+                    pids_db.pop(agent_pid)
+                else:
+                    pids_viaf.append(viaf_pid)
+            return list(pids_db), pids_viaf
+        click.secho(f'ERROR Record class not found for: {agent}', fg='red')
+        return [], []
+
+    @classmethod
+    def get_pids_with_multiple_viaf(cls, record_types=None, verbose=False):
+        """Get agent pids with multiple MEF records.
+
+        :params record_types: Record types (pid_types).
+        :param verbose: Verbose.
+        :returns: pids, multiple pids, missing pids.
+        """
+        multiple_pids = {
+            'gnd_pid': {},
+            'idref_pid': {},
+            'rero_pid': {}
+        }
         progress = progressbar(
-            items=query.scan(),
-            length=query.count(),
+            items=AgentViafSearch()
+            .params(preserve_order=True)
+            .sort({'pid': {'order': 'asc'}})
+            .scan(),
+            length=AgentViafSearch().count(),
             verbose=verbose
         )
         for hit in progress:
             viaf_pid = hit.pid
-            agent_pid = hit.to_dict().get(agent_pid_name)
-            if pids_db.get(agent_pid):
-                pids_db.pop(agent_pid)
-            else:
-                pids_viaf.append(viaf_pid)
-        pids_db = [v for v in pids_db]
-        return pids_db, pids_viaf
-
-    def create_or_update_mef_viaf_record(self, dbcommit=False, reindex=False,
-                                         online=False):
-        """Create or update MEF and VIAF record.
-
-        :param dbcommit: Commit changes to DB.
-        :param reindex: Reindex record.
-        :param online: Try to get VIAF record online.
-        :returns: MEF record, MEF action, VIAF record, VIAF
-        """
-        return None, Action.ERROR, self, False
+            data = hit.to_dict()
+            for agent in multiple_pids:
+                if pid := data.get(agent):
+                    multiple_pids[agent].setdefault(pid, [])
+                    multiple_pids[agent][pid].append(viaf_pid)
+        cleaned_pids = {
+            'gnd_pid': {},
+            'idref_pid': {},
+            'rero_pid': {}
+        }
+        for agent, pids in multiple_pids.items():
+            for pid, viaf_pids in pids.items():
+                if len(viaf_pids) > 1:
```

Comment:
you should use simple `if viaf_pids:` (empty dict / empty list are equivalent to `None`)

Reply:
In this case I will find out if we have more then one element in the list.